Resources/templates/CommonAdmin/generic_actions.php.twig

```diff
@@ -1,42 +1,42 @@
 {% block generic_actions %}
     {{ echo_block("generic_actions") }}
-            {% if builder.actions|default is not empty %}
-                {{ echo_block('pre_generic_actions') }}{{ echo_endblock() }}
-                {% set actionCredentials = builder.actions|mapBy('credentials')|flatten|clean|unique %}
-
-                {% if actionCredentials is not empty %}
-                    {{ echo_set('actionCredentials', echo_twig_arr(actionCredentials), false) }}
-                    {{ echo_if('is_one_admingenerator_granted(actionCredentials)') }}
-                {% endif %}
-
-                {% set excelActions = builder.excelActions is defined ? builder.excelActions : [] %}
-                {% for action in builder.Actions %}
-                    {{ echo_block("generic_action_" ~ action.twigName) }}
-                    {% if action.credentials %}
-                        {{ echo_if_granted(action.credentials) }}
-                    {% endif %}
-
-                    {% if action.name is same as('excel') and excelActions|length %}
+    {% if builder.actions|default is not empty %}
+        {{ echo_block('pre_generic_actions') }}{{ echo_endblock() }}
+        {% set actionCredentials = builder.actions|mapBy('credentials')|flatten|clean|unique %}
+
+        {% if actionCredentials is not empty %}
+            {{ echo_set('actionCredentials', echo_twig_arr(actionCredentials), false) }}
+            {{ echo_if('is_one_admingenerator_granted(actionCredentials, (' ~ builder.ModelClass ~  ' is defined) ? ' ~ builder.ModelClass ~  ' : null )') }}
+        {% endif %}
+
+        {% set excelActions = builder.excelActions is defined ? builder.excelActions : [] %}
+        {% for action in builder.Actions %}
+            {{ echo_block("generic_action_" ~ action.twigName) }}
+            {% if action.credentials %}
+                {{ echo_if_granted(action.credentials, builder.ModelClass) }}
```

Comment:
Here is change. echo_if_granted function generates proper ternary operator, so no need for manually generating expression.